DPGAnalysis/SiStripTools/plugins/EventWithHistoryProducerFromL1ABC.cc

```diff
@@ -104,16 +107,24 @@ void EventWithHistoryProducerFromL1ABC::produce(edm::Event& iEvent, const edm::E
   } else {
     Handle<L1AcceptBunchCrossingCollection> pIn;
     iEvent.getByToken(_l1abccollectionToken, pIn);
-
+    Handle<TCDSRecord> tcds_pIn;
+    iEvent.getByToken(_tcdsRecordToken, tcds_pIn);
+    const auto& tcdsRecord = *tcds_pIn.product();
     // offset computation
 
     long long orbitoffset = 0;
     int bxoffset = 0;
-    if (!_forceNoOffset) {
-      for (L1AcceptBunchCrossingCollection::const_iterator l1abc = pIn->begin(); l1abc != pIn->end(); ++l1abc) {
-        if (l1abc->l1AcceptOffset() == 0) {
-          orbitoffset = (long long)l1abc->orbitNumber() - (long long)iEvent.orbitNumber();
-          bxoffset = l1abc->bunchCrossing() - iEvent.bunchCrossing();
+
+    try {
```

Comment:
as above, I think we should try to avoid `try, catch(...)` statements.

Reply:
Fixed (will push in a while)